### PR TITLE
[WIP] Enhance UnwrapError messaging

### PR DESCRIPTION
**Pull request by noxasaxon**
see #98

### CODE REVIEW

File: src/result/result.py [hunk 1]
```diff
@@ -232,7 +232,16 @@ def unwrap(self) -> NoReturn:
         """
         Raises an `UnwrapError`.
         """
-        raise UnwrapError(self, "Called `Result.unwrap()` on an `Err` value")
+        if isinstance(self.value, Exception):
```
Comment (wbolster) [inline src/result/result.py:235]: this needs to be `BaseException`

File: src/result/result.py [hunk 2]
```diff
@@ -232,7 +232,16 @@ def unwrap(self) -> NoReturn:
...
+            raise UnwrapError(
+                self,
+                f"Called `Result.unwrap()` on an `Err` value: {self.value.__repr__()}",
+            ) from self.value
+        else:
+            raise UnwrapError(
```
Comment (wbolster) [inline src/result/result.py:241]: this is the same code as above; an `exc` variable would help :)

File: src/result/result.py [hunk 3]
```diff
@@ -232,7 +232,16 @@ def unwrap(self) -> NoReturn:
         """
         Raises an `UnwrapError`.
         """
-        raise UnwrapError(self, "Called `Result.unwrap()` on an `Err` value")
+        if isinstance(self.value, Exception):
+            raise UnwrapError(
+                self,
+                f"Called `Result.unwrap()` on an `Err` value: {self.value.__repr__()}",
```
Comment (wbolster) [inline src/result/result.py:238]: f-strings can do `repr()` via the `r` conversion: `f"aaa {value!r} bbb"`

**Review by wbolster — CHANGES_REQUESTED** [review]

File: src/result/result.py [hunk 4]
```diff
@@ -232,7 +232,16 @@ def unwrap(self) -> NoReturn:
...
+            ) from self.value
```
Comment (wbolster) [inline src/result/result.py:239]: this needs test coverage. `exc.__cause__` is what it should look at

Comment (noxasaxon) [reply src/result/result.py:239]: Yeah it's very much a work in progress, I'll remove the draft label when it's ready for review

**Comment by noxasaxon** [comment]
I only made updates to the Err() class' `.unwrap()` & `.except()` methods, because like https://github.com/rustedpy/result/issues/37 says i'm not convinced that the Ok class should even have those methods in the first place if we're striving for the best developer experience with type-checkers.

**Comment by francium** [comment]
Continuing this PR here https://github.com/rustedpy/result/pull/132

**Pull request closed without merging**